debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java

```diff
@@ -96,4 +109,124 @@ public void enableTableCdc(String name) throws SQLException {
+     *
+     * @param tableId - the requested table changes
+     * @param fromLsn - closed lower bound of interval of changes to be provided
+     * @param toLsn  - closed upper bound of interval  of changes to be provided
+     * @param consumer - the change processor
+     * @throws SQLException
+     */
+    public void getChangesForTable(TableId tableId, Lsn fromLsn, Lsn toLsn, ResultSetConsumer consumer) throws SQLException {
+        final String query = GET_ALL_CHANGES_FOR_TABLE.replace(STATEMENTS_PLACEHOLDER, cdcNameForTable(tableId));
+        prepareQuery(query, statement -> {
+            statement.setBytes(1, fromLsn.getBinary());
+            statement.setBytes(2, toLsn.getBinary());
+        }, consumer);
+    }
+
+    /**
+     * Provides all changes recorder by the SQL Server CDC capture process for a set of tables.
+     * 
+     * @param tableIds - the requested tables to obtain changes for
+     * @param fromLsn - closed lower bound of interval of changes to be provided
+     * @param toLsn  - closed upper bound of interval  of changes to be provided
+     * @param consumer - the change processor
+     * @throws SQLException
+     */
+    public void getChangesForTables(TableId[] tableIds, Lsn fromLsn, Lsn toLsn, BlockingMultiResultSetConsumer consumer) throws SQLException, InterruptedException {
+        final String[] queries = new String[tableIds.length];
+
+        int idx = 0;
+        for (TableId tableId: tableIds) {
+            final String query = GET_ALL_CHANGES_FOR_TABLE.replace(STATEMENTS_PLACEHOLDER, cdcNameForTable(tableId));
+            queries[idx++] = query;
+            LOGGER.trace("Getting changes for table {} in range[{}, {}]", tableId, fromLsn, toLsn);
+        }
+        prepareQuery(queries, statement -> {
+            statement.setBytes(1, fromLsn.getBinary());
+            statement.setBytes(2, toLsn.getBinary());
+        }, consumer);
+    }
+
+    /**
+     * Obtain the next available position in the database log.
+     * 
+     * @param lsn - LSN of the current position
+     * @return LSN of the next position in the database
+     * @throws SQLException
+     */
+    public Lsn incrementLsn(Lsn lsn) throws SQLException {
+        final String query = INCREMENT_LSN;
+        return prepareQueryAndMap(query, statement -> {
+            statement.setBytes(1, lsn.getBinary());
+        }, singleResultMapper(rs -> {
+            final Lsn ret = Lsn.valueOf(rs.getBytes(1));
+            LOGGER.trace("Increasing lsn from {} to {}", lsn, ret);
+            return ret;
+        }, "Increment LSN query must return exactly one value"));
+    }
+
+    /**
+     * Map a commit LSN to a point in time when the commit happened.
+     * 
+     * @param lsn - LSN of the commit
+     * @return time when the commit was recorded into the database log
+     * @throws SQLException
+     */
+    public Instant timestampOfLsn(Lsn lsn) throws SQLException {
```

Comment:
That seems rather costly to do another DB round-trip for getting the LSN's timestamp. Isn't there a way to obtain it right when reading from the CDC tables? I'll log an issue to explore that as a follow-up.